contracts/interchain-token-service/src/contract/execute.rs

```diff
@@ -132,3 +136,25 @@ pub fn deregister_its_contract(deps: DepsMut, chain: ChainNameRaw) -> Result<Res
 
     Ok(Response::new().add_event(Event::ItsContractDeregistered { chain }.into()))
 }
+
+pub fn set_chain_config(
+    deps: DepsMut,
+    chain: ChainNameRaw,
+    max_uint: Uint256,
+    max_target_decimals: u8,
+) -> Result<Response, Error> {
+    match state::load_chain_config(deps.storage, &chain).change_context(Error::StorageError)? {
+        Some(_) => bail!(Error::ChainConfigAlreadySet(chain)),
+        None => state::save_chain_config(
+            deps.storage,
+            &chain,
+            max_uint
+                .try_into()
+                .map_err(Report::new)
+                .change_context(Error::InvalidChainMaxUint)?,
```

Comment:
The `StorageError` to me means anything that comes from the `state` module. It is indeed unknow to here.

Reply:
that's not how we use that error across the contracts though

Reply:
https://github.com/axelarnetwork/axelar-amplifier/blob/4adfcefe088e5707a67c5798b2fa5b55f605b7d5/contracts/voting-verifier/src/error.rs#L50
https://github.com/axelarnetwork/axelar-amplifier/blob/4adfcefe088e5707a67c5798b2fa5b55f605b7d5/contracts/multisig-prover/src/error.rs#L96

Reply:
`StateError` reflects anything that comes from the state module, whatever it is. Functions in execute can attach specific info, such as InvalidMaxUint, but it cannot attach anything to `StateError` since it doesn't know anything about it. Now `contract.rs` should be the place to say `FailedToSetChainConfig` since it knows that execute function fails and that's all it knows. 

"We use StorageError across many contracts to just wrap cw_storage_plus errors" if that's the case, then `StorageError` not never be one of the contract errors but the state modules's Error. The better way of using error stack is to attach info at each layer if available, instead of attching anything. Also, the worst possible way of using it is for a function such as `set_chain_config` to have to attach `FailedToSetChainConfig` at every single error it possibly returns.

Reply:
`StateError` is fine with me. And I also agree that `StorageError` should be an error of the state module. Unfortunately I think a lot of our contracts don't define errors for each module, but just errors used across the whole contract. But in ITS hub, I would prefer `StorageError` to live in the state module

Reply:
Anyway, I'll likely re-write the whole thing with the interface change we just talked about. I'll get rid of all the unused errors. There are a couple others I noticed as well.